pkg/scaffold/olm-catalog/csv_updaters.go

```diff
@@ -0,0 +1,279 @@
+
+	return nil
+}
+
+func (us *CSVInstallStrategyUpdate) Apply(csv *olmApi.ClusterServiceVersion) (err error) {
+	// Get install strategy from csv. Default to a deployment strategy if none found.
+	var strat olmInstall.Strategy
+	if csv.Spec.InstallStrategy.StrategyName == "" {
+		csv.Spec.InstallStrategy.StrategyName = olmInstall.InstallStrategyNameDeployment
+		strat = &olmInstall.StrategyDetailsDeployment{}
+	} else {
+		var resolver *olmInstall.StrategyResolver
+		strat, err = resolver.UnmarshalStrategy(csv.Spec.InstallStrategy)
+		if err != nil {
+			return err
+		}
+	}
+
+	switch s := strat.(type) {
+	case *olmInstall.StrategyDetailsDeployment:
+		// Update permissions and deployments.
+		us.updatePermissions(s)
+		us.updateClusterPermissions(s)
+		us.updateDeploymentSpecs(s)
+	default:
+		return fmt.Errorf("install strategy (%v) of unknown type", strat)
+	}
+
+	// Re-serialize permissions into csv strategy.
+	updatedStrat, err := json.Marshal(strat)
+	if err != nil {
+		return err
+	}
+	csv.Spec.InstallStrategy.StrategySpecRaw = updatedStrat
+
+	return nil
+}
+
+func (us *CSVInstallStrategyUpdate) updatePermissions(strat *olmInstall.StrategyDetailsDeployment) {
+	if len(us.Permissions) != 0 {
+		strat.Permissions = us.Permissions
+	}
+}
+
+func (us *CSVInstallStrategyUpdate) updateClusterPermissions(strat *olmInstall.StrategyDetailsDeployment) {
+	if len(us.ClusterPermissions) != 0 {
+		strat.ClusterPermissions = us.ClusterPermissions
+	}
+}
+
+func (us *CSVInstallStrategyUpdate) updateDeploymentSpecs(strat *olmInstall.StrategyDetailsDeployment) {
+	if len(us.DeploymentSpecs) != 0 {
+		strat.DeploymentSpecs = us.DeploymentSpecs
+	}
+}
+
+type CSVCustomResourceDefinitionsUpdate struct {
+	*olmApi.CustomResourceDefinitions
+}
+
+func (store *updaterStore) AddOwnedCRD(yamlDoc []byte) error {
+	newCRDDesc, err := parseCRDDescriptionFromYAML(yamlDoc)
+	if err == nil {
+		store.crdUpdate.Owned = append(store.crdUpdate.Owned, *newCRDDesc)
+	}
+	return err
+}
+
+func (store *updaterStore) AddRequiredCRD(yamlDoc []byte) error {
+	newCRDDesc, err := parseCRDDescriptionFromYAML(yamlDoc)
+	if err == nil {
+		store.crdUpdate.Required = append(store.crdUpdate.Required, *newCRDDesc)
+	}
+	return err
+}
+
+func parseCRDDescriptionFromYAML(yamlDoc []byte) (*olmApi.CRDDescription, error) {
+	newCRD := &apiextv1beta1.CustomResourceDefinition{}
+	if err := yaml.Unmarshal(yamlDoc, newCRD); err != nil {
+		return nil, err
+	}
+
+	crdDesc := &olmApi.CRDDescription{
+		Name:    newCRD.ObjectMeta.Name,
+		Version: newCRD.Spec.Version,
+		Kind:    newCRD.Spec.Names.Kind,
+	}
+	setCRDDisplayNameIfExist(crdDesc, newCRD)
+	setCRDDescriptionIfExist(crdDesc, newCRD)
+
+	return crdDesc, nil
+}
+
+// Annotations for display name and description in CRD metadata should have
+// keys containing strings that match the following patterns if users want
+// CSV's to correctly describe and display their CRD's.
```

Comment:
@estroz I can't see if this was discussed in the design doc or not.
But shouldn't the CRD `displayName` and `displayDescription` fields be user defined as well?

Right now if I generate my CSV once and then modify it to add the display name and description:
```YAML
spec:
  apiservicedefinitions: {}
  customresourcedefinitions:
    owned:
    - kind: AppService
      name: appservices.app.example.com
      version: v1alpha1
      description: An instance of AppService
      displayName: AppService
```
And then rerun `operator-sdk olm-catalog gen-csv` it stomps out the `description` and `displayName`.
I would have expected it to maintain those.

Adding annotations to the crd manifests for these CSV specific fields seems to be a roundabout way of populating these fields. 
I would much rather just edit the generated CSV than have to edit my crd manifest just to satisfy the CSV. WDYT?

Reply:
Yeah I'm fine with allowing the user to solely define these fields.